release-lockfile.spec.md

```diff
@@ -276,14 +286,154 @@ this project depends on.
+link references which **should** be linked with the same *link value*
+**should** be given the same name.
+
+* Required: No
+* Type: String
+* Format: **must** conform to the *Contract Name* format.
+
+
+### The *Link Value* Object
+
+A *Link Value* object is defined to have the following key/value pairs.
+
+
+#### Offset `offset`
+
+The `offset` field defines the location within the corresponding bytecode where
+the `value` for this *link value* was written.  This location is a
+0-indexed offset from the beginning of the binary representation of the
+corresponding bytecode.
+
+* Required: Yes
+* Type: Integer
+* Format: The integer **must** conform to all of the following:
+    * be greater than or equal to zero 
+    * strictly less than the length of the unprefixed hexidecimal representation of the corresponding bytecode.
+
+
+#### Value `value`
+
+The `value` field defines the value which should be written when *linking* the
+corresponding bytecode.
+
+* Required: Yes
+* Type: String
+* Format: One of the following formats.
+
+To reference the address of a *contract instance* from the current release lockfile
+the value should be the name of that *contract instance*.  
+
+* This value **must** be a valid *contract instance* name.
+* The chain definition under which the *contract instance* that this *link value* belongs to must contain this value within its keys.
+* This value **may not** reference the same *contract instance* that this *link value* belongs to.
+
+
+To reference a *contract instance* from a lockfile from somewhere within the
+dependency tree the value is constructed as follows.
+
+* Let `[p1, p2, .. pn]` define a path down the dependency tree.
+* Each of `p1, p2, pn` **must** be valid package names.
+* `p1` **must** be present in keys of the `build_dependencies` for the current release lockfile.
+* For every `pn` where `n > 1`, `pn` **must** be present in the keys of the `build_dependencies` of the lockfile for `pn-1`.
+* The value is represented by the string `<p1>:<p2>:<...>:<pn>:<contract-instance>` where all of `<p1>`, `<p2>`, `<pn>` are valid package names and `<contract-instance>` is a valid contract name.
+* The `<contract-instance>` value **must** be a valid *contract instance* name.
+* Within the release lockfile of the package dependency defined by `<pn>`, all of the following must be satisfiable:
+    * There **must** be *exactly* one chain defined under the `deployments` key which matches the chain definition that this *link value* is nested under.
+    * The `<contract-instance>` value **must** be present in the keys of the matching chain.
+
+To references a static address use the `'0x'` prefixed address as the value.
+Package managers **should not** use this pattern when building releases that
+will be published as open source packages or that are intended to be used
+outside of a closed system.  Package managers **should** require some form of
```

Comment:
Some explanation of this would be nice in the spec. What's the reasoning here?

Reply:
I think I'm actually going to remove this and include it in the "implementers guide"